File: simple_server.py
import http.server
import socketserver
import os

# Set the port
PORT = 5001

# Set up the directory to serve files from
web_dir = os.path.dirname(os.path.abspath(__file__))
os.chdir(web_dir)

# Custom handler to serve specific file types
class CustomHTTPRequestHandler(http.server.SimpleHTTPRequestHandler):
    def do_GET(self):
        # Serve preview.html for root requests
        if self.path == '/' or self.path == '/index.html':
            self.path = '/preview.html'
        return http.server.SimpleHTTPRequestHandler.do_GET(self)
    
    def guess_type(self, path):
        # Add appropriate MIME types
        base, ext = os.path.splitext(path)
        if ext == '.css':
            return 'text/css'
        elif ext == '.html':
            return 'text/html'
        elif ext == '.js':
            return 'application/javascript'
        elif ext == '.svg':
            return 'image/svg+xml'
        elif ext == '.png':
            return 'image/png'
        elif ext == '.jpg' or ext == '.jpeg':
            return 'image/jpeg'
        else:
            return super().guess_type(path)

# Set up and start the server
Handler = CustomHTTPRequestHandler
with socketserver.TCPServer(("0.0.0.0", PORT), Handler) as httpd:
    print(f"Server running at http://0.0.0.0:{PORT}/")
    httpd.serve_forever()

File: simple_server.js
const http = require('http');
const fs = require('fs');
const path = require('path');

const PORT = 5000;

const MIME_TYPES = {
  '.html': 'text/html',
  '.css': 'text/css',
  '.js': 'application/javascript',
  '.json': 'application/json',
  '.png': 'image/png',
  '.jpg': 'image/jpeg',
  '.jpeg': 'image/jpeg',
  '.gif': 'image/gif',
  '.svg': 'image/svg+xml',
};

const server = http.createServer((req, res) => {
  console.log(`Request received: ${req.method} ${req.url}`);
  
  // Serve preview.html for root requests
  let filePath = req.url === '/' ? './preview.html' : '.' + req.url;
  
  // Check if file exists
  fs.access(filePath, fs.constants.F_OK, (err) => {
    if (err) {
      // If file doesn't exist but requested /index.html, serve preview.html
      if (filePath === './index.html') {
        filePath = './preview.html';
      } else {
        res.writeHead(404);
        res.end('404 Not Found');
        return;
      }
    }
    
    // Get the file extension
    const extname = path.extname(filePath);
    const contentType = MIME_TYPES[extname] || 'application/octet-stream';
    
    // Read and serve the file
    fs.readFile(filePath, (err, content) => {
      if (err) {
        if (err.code === 'ENOENT') {
          res.writeHead(404);
          res.end('404 Not Found');
        } else {
          res.writeHead(500);
          res.end(`Server Error: ${err.code}`);
        }
      } else {
        res.writeHead(200, { 'Content-Type': contentType });
        res.end(content, 'utf-8');
      }
    });
  });
});

server.listen(PORT, '0.0.0.0', () => {
  console.log(`Server running at http://0.0.0.0:${PORT}/`);
});